pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -228,14 +237,47 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 		}
 	} else {
 		if contains(instance.ObjectMeta.Finalizers, FinalizerName) {
-			for _, f := range []func(c client.Client, cr *hcov1alpha1.HyperConverged) error{
+			for i, f := range []func(c client.Client, cr *hcov1alpha1.HyperConverged, requestNamespace string) error{
+				// Keep ensuring that we are trying to delete
+				// protected resources first
+				ensureKubeVirtDeleted,
 				ensureCDIDeleted,
 				ensureNetworkAddonsDeleted,
 				ensureKubeVirtCommonTemplateBundleDeleted,
 			} {
-				err = f(r.client, instance)
+				err = f(r.client, instance, request.Namespace)
 				if err != nil {
 					reqLogger.Error(err, "Failed to manually delete objects")
+
+					errT := ErrHCOUninstall
+					errMsg := uninstallHCOErrorMsg
+					switch i {
+					case 0:
+						errT = ErrVirtUninstall
+						errMsg = uninstallVirtErrorMsg + err.Error()
+					case 1:
+						errT = ErrCDIUninstall
+						errMsg = uninstallCDIErrorMsg + err.Error()
+					}
+
+					pod, pod_err := hcoutil.GetPod(r.client, reqLogger)
```

Comment:
My personal opinion is that this should be a `err := r.emitWarning` or somesuch that has all of this process inside (getting the pod, finding the CSV, and emitting the event).

Reply:
I agree and I think we can reuse that in the future